### ISSUE TITLE
False positive: role-with-double-backticks

### ISSUE THREAD

**Issue by hugovk** [sphinx-contrib/sphinx-lint#34]
The valid text has `:all:` at the end of an inline code block (it's a pip option) is reported as invalid.

Small reproducer:

```rst
Only by specifying ``--only-binary :all:`` can you ...

#. Install the requirements file using
   ``pip install --require-hashes --no-deps --only-binary :all:``
```

```console
$ sphinx-lint 1.rst
1.rst:1: role use a single backtick, double backtick found. (role-with-double-backticks)
1.rst:4: role use a single backtick, double backtick found. (role-with-double-backticks)
```

---

To reproduce on a real-life file:

```console
$ wget --quiet https://raw.githubusercontent.com/python/peps/ebefef14d5df408a4a0f8644a454073338fa41aa/pep-0665.rst
$ sphinx-lint pep-0665.rst
pep-0665.rst:190: role use a single backtick, double backtick found. (role-with-double-backticks)
pep-0665.rst:198: role use a single backtick, double backtick found. (role-with-double-backticks)
```

Links to file:
* https://github.com/python/peps/blob/ebefef14d5df408a4a0f8644a454073338fa41aa/pep-0665.rst
* https://raw.githubusercontent.com/python/peps/ebefef14d5df408a4a0f8644a454073338fa41aa/pep-0665.rst
* https://peps.python.org/pep-0665/#secure-by-design

Preview of the valid `:all:` at the end of a code block:

<img width="782" alt="image" src="https://user-images.githubusercontent.com/1324225/172197644-119dea59-f887-413d-a3c1-5bb835552974.png">

**Comment by JulienPalard** [comment]
Thanks for this issue! I resolved it (but not released yet).

**Comment by CAM-Gerlach** [comment]
Thanks!

**Comment by JulienPalard** [comment]
(Released as 0.6.6.)
